src/logic.js

```diff
@@ -368,6 +377,44 @@ define([
         },
         reset: function () {
             this.all = [];
+        },
+        // This is to tell HQ's case summary what is referenced
+        caseReferences: function () {
+            // hq implementation details
+            var ret = {
+                condition: {
+                    answer: null,
+                    question: null,
+                    type: 'always',
+                    operator: null
+                }
+            };
+
+            ret.preload = _.chain(this.all)
+                .filter(function(ref) {
+                    return ref.path.startsWith('#case');
+                })
+                .map(function(ref) {
+                    var info = ref.path.split('/'),
+                        prop = info[2];
+                    if (prop === 'case_name') {
+                        prop = 'name';
+                    }
+                    return [ref.sourcePath.replace(/^#form/, '/data'), prop];
```

Comment:
Every place where this PR matches a prefix like this it should include the `/` after the prefix. So this should be `/^#form\//`.